server.js

```diff
@@ -19,14 +13,115 @@ mongoose.Promise = Promise;
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+const { Schema } = mongoose;
+const songSchema = new Schema({
+  id: Number,
+  trackName: String,
+  artistName: String,
+  genre: String,
+  bpm: Number,
+  energy: Number,
+  danceability: Number,
+  loudness: Number,
+  liveness: Number,
+  valence: Number,
+  length: Number,
+  acousticness: Number,
+  speechiness: Number,
+  popularity: Number
+})
+
+const Song = mongoose.model("Song", songSchema);
+
+// RESET_DB is an environmental variable, write "RESET_DB=true npm run dev" in terminal to reset database
+if (process.env.RESET_DB) {
+  const resetDatabase = async () => {
+    await Song.deleteMany()
+    topMusicData.forEach((singleSong) => {
+    const newSong = new Song (singleSong) 
+    newSong.save() // The .save is what actually saves the song to the database
+    })
+  }
+  // Call a function while declaring it  - extra curriculum (do research)
+  resetDatabase()
+}
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.json(listEndpoints(app));
+});
+
+app.get("/songs", async (req, res) => {
+  const { artistName, genre, danceability } = req.query
+  const response = {
+    success: true,
+    body: {}
+  }
+  // Makes any word search give back search results that includes the word you search for, ex "pop" shows you canadian pop, dance pop etc (works on strings not numbers)
+  const artistRegex = new RegExp(artistName)
+  const genreRegex = new RegExp(genre)
+  // $gt means "greater than" in moongose
+  // if the dancebility value we search for is true return danceability if not return 0
+  const danceabilityQuery = { $gt: danceability ? danceability : 0}
+  try {
+    const searchResultFromDB = await Song.find({artistName: artistRegex, genre: genreRegex, danceability: danceabilityQuery})
+    if (searchResultFromDB) {
+      response.body = searchResultFromDB
+      res.status(200).json(response)
+    } else {
+      response.success = false,
+      res.status(500).json(response)
+    }
+  } catch(e) {
+    res.status(500).json({
+      success: false,
+      body: {
+        message: e
+      }
+    })
+  }
+});
+
+app.get("/songs/id/:id", async (req, res) => {
+  try {
+    // Here it is trying to find a song through its id and returns a different status message depending on if it is found or not
+    const singleSong = await Song.findById(req.params.id)
+    if (singleSong) {
+      res.status(200).json({
+        success: true,
+        body: singleSong
+      })
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "Song not found"
+        }      
+      })
+    }
+  } catch(error) {
+    res.status(500).json({
+      success: false,
+      body: {
+        message: error
+      }
+    })
+  }
 });
```

Comment:
Clean code and easy to read, nice that you added some comments to explain whats going on!